impl/src/metadata.rs

```diff
@@ -19,16 +19,67 @@ use std::{
   string::String,
 };
 
-use anyhow::{anyhow, Result};
+use anyhow::{anyhow, Context, Result};
 
+use cargo_lock::Lockfile;
 use cargo_metadata::MetadataCommand;
-pub use cargo_metadata::{DependencyKind, Metadata, Node, Package, PackageId};
 
+use glob::glob;
+use pathdiff::diff_paths;
+use regex::Regex;
+use rustc_serialize::hex::ToHex;
 use tempfile::TempDir;
+use url::Url;
 
-use rustc_serialize::hex::ToHex;
+pub use cargo_metadata::{DependencyKind, Metadata, Node, Package, PackageId};
```

Comment:
Why pub use?

Reply:
(is this because you wanted to access it from other modules through this module? if so, I'd really prefer you just import it like normal. If it's because you want the library to export this type, I also find that questionable -- I don't think it's a great idea to export other libraries types through our interface)

Reply:
I was thinking of this from the perspective of external crates (things that might import `cargo-raze`). The `RazeMetadata` struct is the source of truth so I thought whatever it was using should be re exported. I thought this was a common pattern since I'd run into it in the `cargo_lock` and `cargo_metadata` crates in how they handle `semver::Version`. Should I still change this?

Reply:
I would think its best for now to not be pub, I dont feel there is a good surface API in cargo-raze for exporting.

Reply:
I think that's worth exploring but I think that should be a focused effort. I've made these not public.

Reply:
Oh, and thanks @GregBowyer for checking out this PR!

I'm particularly interested in hearing feedback on the functionality in here. Would love to get your thoughts/review 😄